_posts/2019-12-10-variational-gaussian-process/vgp.Rmd

```diff
@@ -0,0 +1,549 @@
+description: > 
+ Continuing our tour of applications of TensorFlow Probability (TFP), after Bayesian Neural Networks, Hamiltonian Monte Carlo and State Space Models here we show an example of Gaussian Process Regression. In fact, what we see is a rather "normal" Keras network, defined and trained pretty much the usual way, with TFP's Variational Gaussian Process layer pulling off all the magic. 
+ 
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019vgp
+bibliography: bibliography.bib
+date: 12-10-2019
+categories:
+  - Probability and statistics
+  - Keras
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/kernel_cookbook.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+How do you motivate, or come up with a story around, Gaussian Process Regression on a blog primarily dedicated to deep learning? 
+
+Easy. As demonstrated by seemingly-unavoidable, reliably-recurring twitter "wars" surrounding AI, nothing attracts attention like controversy and antagonism. So, let's go back twenty years and find citations of people saying, "here come Gaussian Processes, we don't need to bother with those finicky, hard to tune neural networks anymore!". And today, here we are; everyone knows _something_ about deep learning but who's heard of Gaussian Processes?
+
+While similar tales tell a lot about history of science and and development of opinions, we prefer a different angle here. In the preface to their 2006 book on _Gaussian Processes for Machine Learning_ [@Rasmussen], Rasmussen and Williams say, referring to the "two cultures" -- the disciplines of statistics and machine learning, respectively: ^[In the book, it's "two communities", not "two cultures"; we choose L. Breiman's [@breiman2001] more punchy expression just ... because.]
+
+> Gaussian process models in some sense bring together work in the two communities. 
+
+In this post, that "in some sense" gets very concrete. We'll see a Keras network, defined and trained the usual way, that has a Gaussian Process layer for its main constituent.
+The task will be "simple" multivariate regression.
+
+As an aside, this "bringing together communities" -- or ways of thinking, or solution strategies -- makes for a good overall characterization of TensorFlow Probability as well.
+^[So far, we have seen uses of TensorFlow Probability for such different applications as [adding uncertainty estimates to neural networks](https://blogs.rstudio.com/tensorflow/posts/2019-11-13-variational-convnet/), [Bayesian model estimation using Hamiltonian Monte Carlo](https://blogs.rstudio.com/tensorflow/posts/2019-05-24-varying-slopes/), or [linear-Gaussian state space models](https://blogs.rstudio.com/tensorflow/posts/2019-06-25-dynamic_linear_models_tfprobability/).]
+
+## Gaussian Processes -- instead of an introduction
```

Comment:
Are you saying that you would usually have an introduction here, but don't? Since we're several paragraphs into the post, this is a little bit ambiguous to me.

Reply:
well, "introduction" as in "introduction to Gaussian Processes", not "introduction overall" ... but you're right, it's ambiguous and kinda superfluous as well, so just gonna throw it away :-)

(also, the other headings are 1-2 words, it didn't fit anyway)